x-pack/libbeat/management/config.go

```diff
@@ -5,32 +5,78 @@
 package management
 
 import (
-	"os"
+	"io"
+	"text/template"
 	"time"
 
-	"github.com/elastic/beats/libbeat/common"
-	"github.com/elastic/beats/libbeat/common/file"
-	"github.com/elastic/beats/libbeat/kibana"
-	"github.com/elastic/beats/libbeat/paths"
-	"github.com/elastic/beats/x-pack/libbeat/management/api"
-
-	"github.com/pkg/errors"
 	"gopkg.in/yaml.v2"
+
+	"github.com/elastic/beats/libbeat/kibana"
 )
 
+// ManagedConfigTemplate is used to overwrite settings file during enrollment
+const ManagedConfigTemplate = `
+
+#========================= Central Management =================================
+
+# Beats is configured under central management, you can define most settings
+# from the Kibana UI. You can update this file to configure the settings that
+# are not supported by Kibana Beats management.
+
+{{.CentralManagementSettings}}
+#================================ General =====================================
+
+# The name of the shipper that publishes the network data. It can be used to group
+# all the transactions sent by a single shipper in the web interface.
+#name:
+
+# The tags of the shipper are included in their own field with each
+# transaction published.
+#tags: ["service-X", "web-tier"]
+
+# Optional fields that you can specify to add additional information to the
+# output.
+#fields:
+#  env: staging
+
+#================================ Logging =====================================
+
+# Sets log level. The default log level is info.
+# Available log levels are: error, warning, info, debug
+#logging.level: debug
+
+# At debug level, you can selectively enable logging only for some components.
+# To enable all selectors use ["*"]. Examples of other selectors are "beat",
+# "publish", "service".
+#logging.selectors: ["*"]
+
+#============================== Xpack Monitoring ===============================
+# {{.BeatName}} can export internal metrics to a central Elasticsearch monitoring
+# cluster.  This requires xpack monitoring to be enabled in Elasticsearch.  The
+# reporting is disabled by default.
+
+# Set to true to enable the monitoring reporter.
+#xpack.monitoring.enabled: false
```

Comment:
I'm not making assumptions on the output settings, I hope that this is one of the first things we convert to central management, in most cases (elasticsearch output) this can be just enabled with it